addons/mrp/models/stock_picking.py

```diff
@@ -36,7 +36,10 @@ def _get_mo_count(self):
         domains = {
             'count_mo_waiting': [('reservation_state', '=', 'waiting')],
             'count_mo_todo': ['|', ('state', 'in', ('confirmed', 'draft', 'progress', 'to_close')), ('is_planned', '=', True)],
-            'count_mo_late': [('date_planned_start', '<', fields.Date.today()), ('state', '=', 'confirmed')],
+            'count_mo_late': ['|', ('delay_alert_date', '!=', False), '&', '|',
+                           ('date_deadline', '<', fields.Date.today()),
+                           ('date_planned_finished', '<', fields.Date.today()),
```

Comment:
I don't see the ('date_planned_finished', '<', fields.Date.today()) in the original filter